### ISSUE TITLE
Check expressions in unmatched mustaches

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#146]
Pretty common autocomplete scenario is start a mustache and immediately start autocompleting it:

```
<!-- html I already wrote ... -->
  <!-- html I'm writing -->{{completeMe.andKeep(completing).from + there
<!-- html I already wrote ... -->
```

After completing the expression, it will be closed with '}}'. Since intelliJ doesn't autoclose mustaches (and won't unless we hack on the angular plugin itself), this is a very common scenario. If we don't resolve the expression as they are typing, autocomplete won't work.

What we can do here is take either
* to the end of the TextNode or
* to the next open {{

and throw parseExpression at it, *allowing* there to be dangling tokens at the end. Need to make sure it ends up in the autocompletable expressions list (and/or the AST we'll be using for autocomplete soon) and bam we'll have autocomplete, with the added benefit of being able to typecheck half complete expressions like `{{half + complete + expression.here()`

**Comment by MichaelRFairhurst** [comment]
@mk13 this is a ticket for autocomplete+, let me know if you have any questions and/or want to take it on!

**Comment by mk13** [comment]
Cool. Goal would be to somehow enable expression checking as soon as the opening moustache ({{) is detected and potentially try and parse before it's closed without throwing unnecessary errors with open moustache.

**Comment by mk13** [comment]
Resolved with pull request 158
